lib/services/reconciler_test.go

```diff
@@ -182,9 +229,9 @@ func TestReconciler(t *testing.T) {
 			// Reconcile and make sure we got all expected callback calls.
 			err = reconciler.Reconcile(context.Background())
 			require.NoError(t, err)
-			require.Equal(t, test.onCreateCalls, onCreateCalls)
-			require.Equal(t, test.onUpdateCalls, onUpdateCalls)
-			require.Equal(t, test.onDeleteCalls, onDeleteCalls)
+			require.ElementsMatch(t, test.onCreateCalls, onCreateCalls)
```

Comment:
Note for reviewers - I've relaxed the conditions on the call tests to allow the callback invocations to be in arbitrary order. I am reasonably sure that the order of callbacks is not defined or important, but will happily listen to anyone that says so.